packages/core/content-manager/server/src/services/document-manager.ts

```diff
@@ -191,85 +173,44 @@ const documentManager = ({ strapi }: { strapi: Core.Strapi }) => {
         .then((result) => result?.entries.at(0));
     },
 
-    async publishMany(entities: Document[], uid: UID.ContentType) {
-      if (!entities.length) {
-        return null;
-      }
-
-      // Validate entities before publishing, throw if invalid
-      await Promise.all(
-        entities.map((document: Document) => {
-          return strapi.entityValidator.validateEntityCreation(
-            strapi.getModel(uid),
-            document,
-            undefined,
-            // @ts-expect-error - FIXME: entity here is unnecessary
-            document
-          );
-        })
-      );
-
-      // Only publish entities without a published_at date
-      const entitiesToPublish = entities
-        .filter((doc: Document) => !doc[PUBLISHED_AT_ATTRIBUTE])
-        .map((doc: Document) => doc.id);
-
-      const filters = { id: { $in: entitiesToPublish } };
-      const data = { [PUBLISHED_AT_ATTRIBUTE]: new Date() };
-      const populate = await buildDeepPopulate(uid);
-
-      // Everything is valid, publish
-      const publishedEntitiesCount = await strapi.db.query(uid).updateMany({
-        where: filters,
-        data,
-      });
-      // Get the updated entities since updateMany only returns the count
-      const publishedEntities = await strapi.db.query(uid).findMany({
-        where: filters,
-        populate,
+    async publishMany(
+      documentIds: Modules.Documents.ID[],
+      uid: UID.ContentType,
+      opts: Omit<DocServiceParams<'publish'>, 'documentId'> = {} as any
+    ) {
+      const publishedEntries = await strapi.db.transaction(async () => {
+        return Promise.all(
+          documentIds.map((id) =>
+            strapi
+              .documents(uid)
+              .publish({ ...opts, documentId: id })
+              .then((result) => result?.entries.at(0))
+          )
+        );
       });
-      // Emit the publish event for all updated entities
-      await Promise.all(
-        publishedEntities!.map((doc: Document) => emitEvent(uid, ENTRY_PUBLISH, doc))
-      );
 
       // Return the number of published entities
-      return publishedEntitiesCount;
+      return { count: publishedEntries.length };
```

Comment:
I am also using this for publishing multiple locales. I think the way you have
this written we would only get the count of published documents and not the
count of published entities? (as you're doing .at(0))

Would it be okay if we did something like